slycot/src/synthesis.pyf

```diff
@@ -550,7 +550,7 @@ subroutine sg03ad(dico,job,fact,trans,uplo,n,a,lda,e,lde,q,ldq,z,ldz,x,ldx,scale
     double precision intent(out), dimension(n), depend(n) :: beta
     integer intent(hide,cache),dimension(n*n),depend(n) :: iwork
     double precision intent(hide,cache),dimension(ldwork) :: dwork
-    integer optional,check(ldwork>=max(1,n)),depend(n) :: ldwork=max(1,max(2*n*n,4*n))
+    integer optional,check(ldwork>=max(1,max(2*n*n,8*n+16))),depend(n) :: ldwork=max(1,max(2*n*n,8*n+16))
```

Comment:
But now I see it, the check formula was a bug too. Would hit you if you call the wrapper with the work array provided and too small (otherwise it is created by the wrapper with the size calculated according to the 2nd formula).